packages/zoe/src/types.js

```diff
@@ -28,122 +27,128 @@
+ * within its vat. The contract and ZCF never have direct access to
+ * the users' payments or the Zoe purses.
+ *
+ * @property {() => Issuer} getInvitationIssuer
+ *
+ * Zoe has a single `invitationIssuer` for the entirety of its
+ * lifetime. By having a reference to Zoe, a user can get the
+ * `invitationIssuer` and thus validate any `invitation` they receive
+ * from someone else. The mint associated with the invitationIssuer
+ * creates the ERTP payments that represent the right to interact with
+ * a smart contract in particular ways.
+ *
+ * @property {Install} install
+ * @property {StartInstance} startInstance
+ * @property {Offer} offer
+ * @property {(instance: Instance) => Object} getPublicFacet
+ * @property {(instance: Instance) => IssuerKeywordRecord} getIssuers
+ * @property {(instance: Instance) => BrandKeywordRecord} getBrands
+ * @property {(instance: Instance) => Object} getTerms
+ */
+
+/**
+ * @typedef {Object} Installation
+ * @property {() => SourceBundle} getBundle
+ */
+
+/**
+ * @callback Install
+ * @param {SourceBundle} bundle
+ * @returns {Promise<Installation>}
+ *
  * Create an installation by safely evaluating the code and
- * registering it with Zoe. Returns an installationHandle.
+ * registering it with Zoe. Returns an installation.
+ */
+
+/**
+ * @typedef {Object} UserSeat
+ * @property {() => Promise<Allocation>} getCurrentAllocation
+ * @property {() => Promise<ProposalRecord>} getProposal
+ * @property {() => Promise<PaymentPKeywordRecord>} getPayouts
+ * @property {(keyword: Keyword) => Promise<Payment>} getPayout
+ * @property {() => Promise<OfferResult>} getOfferResult
+ * @property {() => void=} exit
+ * @property {() => Promise<boolean>} hasExited
+ * @property {() => Promise<Notifier>} getNotifier
+ *
+ * @typedef {any} OfferResult
+ */
+
+/**
+ * @callback Offer
+ * @param {Invitation|Promise<Invitation>} invitation
+ * @param {Proposal=} proposal
+ * @param {PaymentKeywordRecord=} paymentKeywordRecord
+ * @returns {Promise<UserSeat>} seat
+ *
+ * To redeem an invite, the user normally provides a proposal (their
+ * rules for the offer) as well as payments to be escrowed by Zoe.  If
+ * either the proposal or payments would be empty, indicate this by
+ * omitting that argument or passing undefined, rather than passing an
+ * empty record.
+ *
+ * The proposal has three parts: `want` and `give` are used by Zoe to
+ * enforce offer safety, and `exit` is used to specify the particular
+ * payout-liveness policy that Zoe can guarantee. `want` and `give`
+ * are objects with keywords as keys and amounts as values.
+ * `paymentKeywordRecord` is a record with keywords as keys, and the
+ * values are the actual payments to be escrowed. A payment is
+ * expected for every rule under `give`.
+ */
+
+/**
+ * @typedef {Object} StartInstanceResult
+ * @property {Record<string, Function>} creatorFacet
+ * @property {Record<string, Function>} publicFacet
```

Comment:
Usually the public facet is typed `{Object}`. Here it is `{Record<String,Function>}`. One place it is `{PublicFacet}` where `PublicFacet` is not mentioned anywhere else.